### PR TITLE
feat(plugin): Add cross-platform support

### PR DESCRIPTION
**Pull request by simar7**
See test run here: https://github.com/simar7/trivy-aws/actions/runs/9461161139/job/26061342531

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,6 +40,14 @@ update-aws-deps:
 	@grep aws-sdk-go-v2 go.mod | grep -v '// indirect' | sed 's/^[ [[:blank:]]]*//g' | sed 's/[[:space:]]v.*//g' | xargs go get
 	@go mod tidy
 
-.PHONY: bundle
-bundle:
-	tar -cvzf trivy-aws.tar.gz plugin.yaml trivy-aws LICENSE
\ No newline at end of file
+.PHONY: bundle-linux
+bundle-linux:
+	tar -cvzf trivy-aws-linux.tar.gz plugin.yaml trivy-aws-linux-amd64 trivy-aws-linux-arm64 LICENSE
```
Comment (nikpivkin) [inline Makefile:45]: Why not use an archive for each architecture and OS?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,24 +1,49 @@
...
   push:
     tags:
       - "v*"
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
```
Comment (nikpivkin) [inline .github/workflows/release.yml:13]: Cross compilation is possible on a single machine. We can only use linux.

File: Makefile [hunk 3]
```diff
@@ -4,7 +4,27 @@ test:
...
+build-darwin-amd64:
+	CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 go build -ldflags "-s -w" -o trivy-aws-darwin-amd64 ./cmd/trivy-aws/main.go
+
+.PHONY: build-darwin-arm64
+build-darwin-arm64:
+	CGO_ENABLED=0 GOOS=darwin GOARCH=arm64 go build -ldflags "-s -w" -o trivy-aws-darwin-arm64 ./cmd/trivy-aws/main.go
+
+.PHONY: build-windows-amd64
+build-windows-amd64:
+	CGO_ENABLED=0 GOOS=windows GOARCH=amd64 go build -ldflags "-s -w" -o trivy-aws-windows-amd64 ./cmd/trivy-aws/main.go
```
Comment (nikpivkin) [inline Makefile:27]:
```suggestion
PLATFORMS = linux/amd64 linux/arm64 darwin/amd64 darwin/arm64 windows/amd64
OUTPUTS = $(patsubst %,%/trivy-aws,$(PLATFORMS))

build: $(OUTPUTS)

# os/arch/trivy-aws
%/trivy-aws:
	@mkdir -p $(dir $@); \
	GOOS=$(word 1,$(subst /, ,$*)); \
	GOARCH=$(word 2,$(subst /, ,$*)); \
	CGO_ENABLED=0 GOOS=$$GOOS GOARCH=$$GOARCH go build -ldflags "-s -w" -o trivy-aws-$$GOOS-$$GOARCH ./cmd/trivy-aws/main.go; \
	tar -cvzf trivy-aws-$$GOOS-$$GOARCH.tar.gz plugin.yaml trivy-aws-$$GOOS-$$GOARCH LICENSE
```

**Comment by simar7** [comment]
@nikpivkin updated as per comments. Thanks for the feedback!

File: Makefile (hunk 3 above)
Comment (simar7) [reply Makefile]: nice!

File: .github/workflows/release.yml (hunk 2 above)
Comment (simar7) [reply .github/workflows/release.yml]: Right - I tried this locally as well although it works, cross compilation takes a while. It's not a big deal.

File: plugin.yaml [hunk 4]
```diff
@@ -2,11 +2,34 @@ name: aws-scan
...
+version: "0.11.0"
 usage: AWS Trivy plugin
 description: |-
   Trivy AWS plugin for the Trivy security scanner
   https://github.com/aquasecurity/trivy-aws
 platforms:
-  - uri: https://github.com/aquasecurity/trivy-aws/releases/download/v0.10.0/trivy-aws.tar.gz
-    bin: ./trivy-aws
\ No newline at end of file
+  - selector:
```
Comment (nikpivkin) [inline plugin.yaml:11]: The selectors should be updated. Now a different archive is created for each architecture.

Comment (simar7) [reply plugin.yaml:11]: ah thanks, missed that. I updated it.

**Review by nikpivkin — APPROVED** [review]

**Pull request merged**